### ISSUE TITLE
Problem found while using DoT service: Destination Port Unreachable

### ISSUE THREAD

**Issue by ha-ku** [ookangzheng/blahdns#111]
I tried to set the DoT server as private dns server on my android device but it didn't work. So i tried to ping the dot server and get:
`From jp1.blahdns.com (45.32.55.94) icmp_seq=1 Destination Port Unreachable`
`From de1.blahdns.com (159.69.198.101) icmp_seq=1 Destination Port Unreachable`
`From fi1.blahdns.com (95.216.212.177) icmp_seq=1 Destination Port Unreachable`

Then I tried to traceroute and get: 
`1  /*my computer*/ (xxx.xxx.xxx.xxx)  0.282 ms  0.319 ms  0.318 ms`
` 2  192.168.1.1 (192.168.1.1)  2.753 ms  2.763 ms  2.761 ms`
` 3  192.168.4.4 (192.168.4.4)  3.548 ms  3.533 ms  3.539 ms`
` 4  100.xxx.xxx.xxx (100.xxx.xxx.xxx)  68.942 ms  68.919 ms  68.931 ms`
` 5  182.xxx.xxx.xxx (182.xxx.xxx.xxx)  5.001 ms  5.597 ms  5.587 ms`
` 6  171.xxx.xxx.xxx (171.xxx.xxx.xxx)  6.488 ms  4.995 ms 171.xxx.xxx.xxx (171.xxx.xxx.xxx)  7.197 ms`
` 7  202.97.36.221 (202.97.36.221)  39.386 ms 202.97.36.225 (202.97.36.225)  38.684 ms 202.97.36.221 (202.97.36.221)  41.344 ms`
` 8  202.97.94.146 (202.97.94.146)  52.984 ms 202.97.94.130 (202.97.94.130)  40.013 ms  40.009 ms`
` 9  202.97.94.90 (202.97.94.90)  54.918 ms 202.97.94.94 (202.97.94.94)  53.975 ms 202.97.12.45 (202.97.12.45)  56.723 ms`
`10  202.97.94.14 (202.97.94.14)  147.806 ms  147.773 ms  148.259 ms`
`11  ae-1.r30.tokyjp05.jp.bb.gin.ntt.net (129.250.2.157)  81.073 ms ae-2.r31.tokyjp05.jp.bb.gin.ntt.net (129.250.2.155)  95.660 ms ae-1.r30.tokyjp05.jp.bb.gin.ntt.net (129.250.2.157)  81.069 ms`
`12  ae-2.r02.tokyjp05.jp.bb.gin.ntt.net (129.250.3.22)  147.686 ms ae-3.r02.tokyjp05.jp.bb.gin.ntt.net (129.250.3.28)  151.243 ms  151.302 ms`
`13  ce-0-14-0-1.r02.tokyjp05.jp.ce.gin.ntt.net (120.88.54.98)  208.942 ms  226.747 ms  210.172 ms`
`14  * * *`
`15  * * *`
`16  * * *`
`17  * * *`
`18  * * *`
`19  * * *`
`20  * * *`
`21  * * *`
`22  * * *`
`23  * * *`
`24  * * *`
`25  * * *`
`26  * * *`
`27  * * *`
`28  * * *`
`29  * * *`
`30  * * *`
It seems like the ping package did reach japanese network but somehow got lost.
By the way, I have no problem connecting to the DoH service(https://doh2.blahdns.com/dns-query), but android doesn't support that. 
Anyone has idea about what happened ?

**Comment by ookangzheng** [comment]
From what I noticed that `doh1` and `doh2` is using CDN. (will be more stable)

Which country are you from? cuz maybe ISP from your country may block blahdns IP (censorship)
You can do `mtr jp1.dadd.icu` again.

**Comment by ha-ku** [comment]
> `doh1` and `doh2` is over CDN will be more stable than DoT without a doubt.
> 
> Which country are you from? cuz maybe ISP from your country may block blahdns IP (censorship via firewall).
> 
> You can do `mtr jp1.dadd.icu` again.

Here's the result: 
`Start: 2020-07-08T11:40:57+0800`
`HOST: LAPTOP-XXXXXXXX                    Loss%   Snt   Last   Avg  Best  Wrst StDev`
`  1.|-- LAPTOP-XXXXXXXX.mshome.ne  0.0%   100    0.3   0.3   0.2   0.5   0.1`
`  2.|-- 192.168.1.1                                   0.0%   100    1.7   1.7   1.1   3.9   0.4`
`  3.|-- 192.168.4.4                                   0.0%   100    1.8   2.0   1.3   8.6   1.1`
`  4.|-- 100.xxx.xxx.xxx                              0.0%   100    3.2   5.8   3.1  85.9   9.6`
`  5.|-- 182.xxx.xxx.xxx                              63.0%   100    4.1   8.7   3.8  27.1   5.7`
`  6.|-- 171.xxx.xxx.xxx                              0.0%   100    6.5   6.9   3.3 110.0  10.5`
`  7.|-- 202.97.36.225                               0.0%   100   40.8  40.4  37.9  51.4   2.1`
`  8.|-- 202.97.94.146                               55.0%   100   40.2  41.6  39.3  55.4   3.0`
`  9.|-- 202.97.94.98                                 10.0%   100   38.2  46.0  35.7  58.2   4.2`
` 10.|-- 202.97.60.214                              8.0%   100  100.4 110.2  94.7 244.5  29.0`
` 11.|-- ae-2.r31.tokyjp05.jp.bb.g            12.0%   100   96.3 100.6  95.8 203.6  17.3`
` 12.|-- ae-3.r02.tokyjp05.jp.bb.g            14.0%   100  145.7 145.9 134.7 150.8   2.0`
` 13.|-- ce-0-14-0-1.r02.tokyjp05.           3.0%   100  204.3 209.7 203.4 286.7  14.0`
` 14.|-- ???                                               100.0   100    0.0   0.0   0.0   0.0   0.0`
` 15.|-- ???                                               100.0   100    0.0   0.0   0.0   0.0   0.0`
` 16.|-- ???                                               100.0   100    0.0   0.0   0.0   0.0   0.0`
` 17.|-- jp1.blahdns.com                          6.0%   100  209.7 209.3 207.8 213.0   0.7`

As I mentioned above, it looks like the package did reach Japan. Thus I guess the problem may not arise from the ISP blocking (otherwise the package would be lost earlier, I suppose). By the way, I live in mainland China now.
I tried the private DNS on my android device again. It still doesn't work. Doing `ping jp1.dadd.icu` also has the `Destination Port Unreachable` response.

**Comment by ookangzheng** [comment]
Sorry for that, I realized that my firewall had block entire China IP to prevent DDos attack. 
If you have fixed IP, I can manually add into `allowlist`.

**Comment by ha-ku** [comment]
OK, that's reasonable.
Seems like it's not easy to get a fixed IP in China. But with some effort, I managed to apply DoH to my android device as an alternative solution. 
Thanks for helping! 😉

**Comment by ookangzheng** [comment]
What kind of mobile operating system do you use? I can suggest other alternative performance app 😄 .

**Comment by ha-ku** [comment]
I'm using MIUI, which is a custom OS based on android (accurately, android 10). I'm using v2rayNG to perform proxy and DoH query now.

**Comment by ookangzheng** [comment]
Ah, so you do have a server right, If so, I think would be nice use dnscrypt-proxy with DoH

**Comment by ha-ku** [comment]
Yes, I actually do have a server. But since the server isn't locate in China, the network latency cannot be ignored. Thus maybe its not a good idea to use it for DNS query.  In fact, I have tried to proxy the DNS query, but it turned out to be too slow.
I'm not sure about what "performance via DoH" means... If you means the DoH services I've tried, it include cloudflare-dns and several other small DoH service.
